File: c++/11501.cpp
#include <iostream>

using namespace std;

int arr[1000000];

int main() {
  freopen("example.txt", "r", stdin);
  ios_base::sync_with_stdio(0);
  cin.tie(0);

  int T;
  cin >> T;

  while (T--) {
    int N;
    cin >> N;
    for (int i = 0; i < N; i++) {
      cin >> arr[i];
    }

    int maxV = 0;
    long long ans = 0;

    for (int i = N - 1; i >= 0; i--) {
      if (maxV < arr[i])
        maxV = arr[i];
      else if (maxV > arr[i])
        ans += maxV - arr[i];
    }

    cout << ans << '\n';
  }

  return 0;
}

File: js/11501.js
const fs = require('fs');

const inputs = fs
  .readFileSync('example.txt')
  .toString()
  .trim()
  .split('\n')
  .map((e) => e.split(' ').map(Number));

let idx = 0;
const getLine = () => inputs[idx++];

let [T] = getLine();

const answer = [];

while (T--) {
  const [N] = getLine();
  const arr = getLine();
  let ans = 0;
  let maxV = 0;

  for (let i = N - 1; i >= 0; i--) {
    if (arr[i] > maxV) maxV = arr[i];
    else if (arr[i] < maxV) ans += maxV - arr[i];
  }

  answer.push(ans);
}

console.log(answer.join('\n'));
